### PR TITLE
[doc]Added advice for production env variables

### PR DESCRIPTION
**Pull request by ObjectiveTruth**
# Pull Request Checklist

* [ ] Have you read [How to write the perfect pull request](https://github.com/blog/1943-how-to-write-the-perfect-pull-request)?
* [ ] Have you read through the [contributor guidelines](https://www.playframework.com/contributing)?
* [ ] Have you signed the [Lightbend CLA](https://www.lightbend.com/contribute/cla)?
* [ ] Have you [squashed your commits]? (Optional, but makes merge messages nicer.)
* [ ] Have you added copyright headers to new files?
* [ ] Have you checked that both Scala and Java APIs are updated?
* [ ] Have you updated the documentation for both Scala and Java sections?
* [ ] Have you added tests for any changed functionality?

# Helpful things

## Fixes

Fixes #xxxx

## Purpose

What does this PR do?

## Background Context

Why did you take this approach?

## References

Are there any relevant issues / PRs / mailing lists discussions?

Had an issue that seemed silly, so I updated the docs with it explicitly. Basically you can't name your Env variables the same as the variables

### CODE REVIEW

**Comment by typesafehub-validator** [comment]
Hi @ObjectiveTruth,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Typesafe Contributors License Agreement:

[http://www.lightbend.com/contribute/cla](http://www.lightbend.com/contribute/cla)

File: documentation/manual/working/commonGuide/production/ProductionConfiguration.md [hunk 1]
```diff
@@ -104,6 +104,17 @@ my.key = ${?MY_KEY_ENV}
 
 Here, the override field `my.key = ${?MY_KEY_ENV}` simply vanishes if there's no value for `MY_KEY_ENV`, but if you set an environment variable `MY_KEY_ENV` for example, it would be used.
 
+Because you can reference variables from withing other variables, ensure you don't name your environmental variable the same as the field name.
```
Comment (gmethvin) [inline documentation/manual/working/commonGuide/production/ProductionConfiguration.md]: s/withing/within/

**Comment by gmethvin** [comment]
Is it worth mentioning this other special case? https://github.com/typesafehub/config#note-about-resolving-substitutions-in-referenceconf-and-applicationconf

**Comment by ObjectiveTruth** [comment]
>Is it worth mentioning this other special case? https://github.com/typesafehub/config#note-about-resolving-substitutions-in-referenceconf-and-applicationconf

I don't think so, because that's more a *best practice* than a *gotcha*. If the user is expecting some programmatic operation in their `*.conf` file, that probably means they don't understand what the `*.conf` files are used for, or will look up more info themselves

**Pull request merged**